server/linter/rules/rules_registry.go

```diff
@@ -0,0 +1,52 @@
+package rules
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kubeshop/tracetest/server/linter/analyzer"
+	"github.com/kubeshop/tracetest/server/model"
+)
+
+type Rule interface {
+	Id() string
+	Evaluate(context.Context, model.Trace, analyzer.LinterRule) (analyzer.RuleResult, error)
+}
+
+type RuleRegistry interface {
```

Comment:
Who depends on this interface? can the dependees directly reference a struct? What benefits do we get from this interface?

If a client depends on this interface, it will be requiring this package. Given that we only support one implementation of the `RuleRegistry`, and it's in memory without any dependencies, doesn't look like we would need to mock it, for example, or replace it by another implementation.

Reply:
Makes sense, I think the best case is just to remove the interface